src/NUnitFramework/framework/Internal/RuntimeFramework.cs

```diff
@@ -226,21 +219,18 @@ public static RuntimeFramework CurrentFramework
         /// <summary>
         /// The framework version for this runtime framework
         /// </summary>
-        public Version FrameworkVersion { get; private set; }
+        public Version FrameworkVersion { get; private set; } = null!;
```

Comment:
This is pure lying, saying that null is not null.

There are three ways to solve this issue:
1.  Suppress the message in the constructor:
```csharp
#pragma warning disable CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
        public RuntimeFramework( RuntimeType runtime, Version version)
#pragma warning restore CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
```

2. Do it declarative way, by adding `MemberNotNull` attributes to the two `InitFrom... Version` methods. This however requires include a reference to the `nullable` nuget package and increase the `LangVersion` to 9.0.
```csharp
        [MemberNotNull(nameof(FrameworkVersion))]
        [MemberNotNull(nameof(ClrVersion))]
        private void InitFromFrameworkVersion(Version version)
```

3.  Rewrite to code so that the two properties are set in the constructor itself.
```csharp
            (FrameworkVersion, ClrVersion) = version.Build < 0 ?
                InitFromFrameworkVersion(version) :
                InitFromClrVersion(version);
```
With a matching declaration and updated code for:
```csharp
        private (Version frameworkVersion, Version clrVersion) InitFromFrameworkVersion(Version version)
        {
            Version frameworkVersion, clrVersion;

            frameworkVersion = clrVersion = version;
```
etc.

Reply:
updated with option 2